cli/cmds.go

```diff
@@ -424,10 +306,253 @@ func versionCmd(cmd *cobra.Command, args []string) {
 	tfCmd.Stdout = stdout
 	tfCmd.Stderr = stderr
 	if err := tfCmd.Run(); err != nil {
-		common.Log.Warnf("Error running terraform: %s", err)
+		log.Warnf("Error running terraform: %s", err)
 		return
 	}
 	terraformVersion := strings.SplitN(stdout.String(), "\n", 2)[0]
 	// nolint errcheck
 	fmt.Println("  + " + terraformVersion)
 }
+
+type textBlockWriter struct {
+	writer io.Writer
+}
+
+func (w textBlockWriter) Write(index, startLine, endLine int, text string) {
+	fmt.Fprint(w.writer, c.Sprintf("\n<white>#######</> <cyan>B%d</><darkGray> @ #%d</>\n", index, endLine))
+	fmt.Fprint(w.writer, text)
+}
+
+func (w textBlockWriter) Close() error { return nil }
+
+type zeroTerminatedBlockWriter struct {
+	writer io.Writer
+}
+
+func (w zeroTerminatedBlockWriter) Write(index, startLine, endLine int, text string) {
+	fmt.Fprint(w.writer, text)
+	fmt.Fprint(w.writer, "\x00")
+}
+
+func (w zeroTerminatedBlockWriter) Close() error { return nil }
+
+type Block struct {
+	StartLine int    `json:"start_line"`
+	EndLine   int    `json:"end_line"`
+	Text      string `json:"text"`
+}
+
+type Output struct {
+	BlockCount int     `json:"block_count"`
+	Blocks     []Block `json:"blocks"`
+}
+
+type jsonBlockWriter struct {
+	writer io.Writer
+	data   Output
+}
+
+func (w *jsonBlockWriter) Write(index, startLine, endLine int, text string) {
+	w.data.BlockCount++
+	w.data.Blocks = append(w.data.Blocks, Block{
+		StartLine: startLine,
+		EndLine:   endLine,
+		Text:      text,
+	})
+}
+
+func (w jsonBlockWriter) Close() error {
+	encoder := json.NewEncoder(w.writer)
+	return encoder.Encode(w.data)
+}
+
+func findBlocksInFile(fs afero.Fs, log *logrus.Logger, filename string, verbose, zeroTerminated, jsonOutput bool, stdin io.Reader, stdout, stderr io.Writer) error {
```

Comment:
This is starting to feel a bit unwieldy, with all the flags being passed as parametrs. At some point, a command object containing the configuration should probably be created.

Reply:
agreed, but that is for another pr!